tests/virt_autotest/virsh_external_snapshot.pm

```diff
@@ -0,0 +1,59 @@
+# SUSE's openQA tests
+#
+# Copyright (C) 2019 SUSE LLC
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, see <http://www.gnu.org/licenses/>.
+
+# Summary: Test VM external snapshot using virsh (create - restore - delete)
+# Maintainer: Leon Guo <xguo@suse.com>
+#
+
+use base "consoletest";
+use strict;
+use warnings;
+use testapi;
+use set_config_as_glue;
+
+sub run {
+    #Snapshots are supported on KVM VM Host Servers only
+    return unless check_var("REGRESSION", "qemu-hypervisor") || check_var("SYSTEM_ROLE", "kvm");
+
+    my $wait_script                 = "30";
+    my $get_vm_hostnames_inactive   = "virsh list --inactive | grep sles | awk \'{print \$2}\'";
+    my $vm_hostnames_inactive       = script_output($get_vm_hostnames_inactive, $wait_script, type_command => 0, proceed_on_failure => 0);
+    my @vm_hostnames_inactive_array = split(/\n+/, $vm_hostnames_inactive);
+
+    foreach (@vm_hostnames_inactive_array)
+    {
+        assert_script_run("virsh start $_");
+    }
```

Comment:
This could be one-line of this sort:
```suggestion
    assert_script_run("virsh start $_") foreach (@vm_hostnames_inactive_array);
```
Also in `tests/virt_autotest/set_config_as_glue.pm`.

Reply:
Cool, made the changed as your mentioned.

Reply:
code changed, verification run:
http://10.67.18.132/tests/258